### PR TITLE
refactor(in-house-liquidity): Refactor functions for better type checking and logic consolidation

### PR DESCRIPTION
**Pull request by jh2oman**
### Description

Based on [this comment](https://github.com/valora-inc/wallet/pull/1881#discussion_r794927093) from @cajubelt

In general, bringing more logic and error handling into the in-house-liquidity functions to make them easier to work with. For instance, now they return specific types as opposed to a generic Response object that callers must interpret.

### Other changes

_Describe any minor or "drive-by" changes here.

### Tested

`yarn test` and manually walking through the flows to ensure they still work

### How others should test

No new features or fixes, just a refactor so there should be no need.

### Related issues

- Fixes #[issue number here]

### Backwards compatibility

_Brief explanation of why these changes are/are not backwards compatible._

### CODE REVIEW

File: packages/mobile/src/in-house-liquidity/index.ts [hunk 1]
```diff
@@ -4,9 +4,13 @@ import KeyEncoder from 'key-encoder'
...
 import { hexToBuffer, trimLeading0x } from '@celo/utils/lib/address'
 
-interface DeleteFinclusiveBankAccountParams {
-  accountMTWAddress: string
-  dekPrivate: string
+const keyEncoder = new KeyEncoder('secp256k1')
+interface RequiredParams {
+  accountMTWAddress: string | null
+  dekPrivate: string | null
+}
```
Comment (cajubelt) [inline packages/mobile/src/in-house-liquidity/index.ts:11]: I don't think required params should be allowed to be null-- seems like the type and the name are at odds

File: packages/mobile/src/in-house-liquidity/index.ts [hunk 2]
```diff
@@ -69,10 +77,14 @@ export const getFinclusiveBankAccounts = async ({
       },
     },
   })
+  if (!response.ok) {
+    throw new Error(`IHL GET /account/bank-account failure status ${response.status}`)
+  }
+  const { bankAccounts } = await response.json()
+  return bankAccounts
```
Comment (cajubelt) [inline packages/mobile/src/in-house-liquidity/index.ts:84]: nice ✅

Comment (cajubelt) [reply packages/mobile/src/in-house-liquidity/index.ts:84]: moving the return type assignments and the response status checks to this layer is a good improvement

**Review by cajubelt — COMMENTED** [review]
Thanks for doing this. I like how you moved the response status checks and the response type assignments to the IHL request layer. 

I left comments about this but to summarize, I'd prefer if the MTW and DEK were still required params in the IHL request layer, since those functions will not work without them. Happy to discuss more if you want.

File: packages/mobile/src/in-house-liquidity/index.ts [hunk 3]
```diff
@@ -262,10 +287,14 @@ export const signAndFetch = async ({
...
-  dekPrivate: string
-}): Promise<string> => {
+}: RequiredParams): Promise<string> => {
+  if (!accountMTWAddress) {
+    throw new Error('Cannot call IHL because accountMTWAddress is null')
+  }
+
+  if (!dekPrivate) {
+    throw new Error('Cannot call IHL because dekPrivate is null')
+  }
```
[the remark this answers is not in the document]
Comment (jh2oman) [reply packages/mobile/src/in-house-liquidity/index.ts:297]: > By making the parameters optional in the input type, it makes it seem like a request with a null MTW address (for instance) is valid, but then a runtime error will occur-- which makes the typing a lot less useful

Ah, thank you. I definitely see the logic in that.

[the remark this answers is not in the document]
Comment (jh2oman) [reply packages/mobile/src/in-house-liquidity/index.ts:297]: > Maybe there's a typescript-friendly way to abstract the null checks into a helper to achieve the same effect

I think i just thought of one! I'll move it out and update the PR with my changes

File: packages/mobile/src/account/BankAccounts.test.tsx [hunk 4]
```diff
@@ -26,16 +26,9 @@ const MOCK_BANK_ACCOUNTS = {
...
-  ok: true,
-  json: () => Promise.resolve(MOCK_BANK_ACCOUNTS),
-})
-const mockDeleteResponse = Promise.resolve({
-  ok: true,
-})
 jest.mock('src/in-house-liquidity', () => ({
-  getFinclusiveBankAccounts: jest.fn(() => mockGetResponse),
-  deleteFinclusiveBankAccount: jest.fn(() => mockDeleteResponse),
+  getFinclusiveBankAccounts: jest.fn(() => Promise.resolve(MOCK_BANK_ACCOUNTS)),
```
[the remark this answers is not in the document]
Comment (jh2oman) [reply packages/mobile/src/account/BankAccounts.test.tsx:29]: They are actually not quite the same thing and the test fails trying to run it your suggested way. I think it has to do with the jest.useRealTimers that we have in most of these components (which I think are needed for the useAsync hooks we use)

File: packages/mobile/src/account/SyncBankAccountScreen.test.tsx [hunk 5]
```diff
@@ -10,18 +10,14 @@ import { Screens } from 'src/navigator/Screens'
...
+
 const mockProps = getMockStackScreenProps(Screens.SyncBankAccountScreen, {
   publicToken: mockPublicToken,
 })
 
 jest.mock('src/in-house-liquidity', () => ({
-  createFinclusiveBankAccount: jest.fn(() => mockFinclusiveBankAccountReponse),
-  exchangePlaidAccessToken: jest.fn(() => mockPlaidAccessTokenResponse),
+  createFinclusiveBankAccount: jest.fn(() => Promise.resolve()),
+  exchangePlaidAccessToken: jest.fn(() => Promise.resolve(mockAccessToken)),
```
[the remark this answers is not in the document]
Comment (jh2oman) [reply packages/mobile/src/account/SyncBankAccountScreen.test.tsx:21]: not quite the same per other comment

**Review by cajubelt — APPROVED** [review]
🥳

**Pull request merged**